packages/amplify-provider-awscloudformation/src/aws-utils/aws-cfn.js

```diff
@@ -73,99 +82,115 @@ class CloudFormation {
     });
   }
 
-  readStackEvents(stackName) {
-    // Stores all the events displayed to the user (useful to do a diff with new Events)
-    let allShownEvents = [];
-    const self = this;
-    const cfnModel = this.cfn;
-    const params = {
-      StackName: stackName,
-    };
-    let nestedStackQueue = [];
-
-    this.pollForEvents = setInterval(() => {
-      cfnModel.describeStackEvents(params, (err, data) => {
-        if (err) {
-          console.log(err);
-          /* Not throwing error out here since this is an asynchornous event
-          running in the background */
-        } else {
-          let stackEvents = data.StackEvents;
-          stackEvents = stackEvents
-            .filter(event => self.eventStartTime < new Date(event.Timestamp));
-
-          if (allShownEvents.length === 0) {
-            // Showing events for the first time to the user
-            if (stackEvents.length > 0) {
-              showEvents(stackEvents);
-              allShownEvents = stackEvents;
-            }
-          }
+  collectStackErrors(stackName) {
+    // add root stack to see the new stacks
+    this.readStackEvents(stackName);
+    // wait for the poll queue to drain
+    return new Promise((resolve) => {
+      this.pollQueue.once('empty', () => {
+        const failedStacks = this.stackEvents.filter(ev =>
+          CNF_ERROR_STATUS.includes(ev.ResourceStatus));
+        try {
+          const trace = this.generateFailedStackErrorMsgs(failedStacks);
+          console.log(`\n\n${chalk.reset.red.bold('Following resources failed')}\n`);
+          trace.forEach((t) => {
+            console.log(t);
+            console.log('\n');
+          });
+          resolve();
+        } catch (e) {
+          Promise.reject(e);
+        }
+      });
+    });
+  }
 
-          let newEvents = stackEvents;
-          const nestedStackEventPromises = [];
+  generateFailedStackErrorMsgs(eventsWithFailure) {
+    const { envName } = this.context.amplify.getEnvInfo();
+    const stackTrees = eventsWithFailure
+      .filter(stack => stack.ResourceType !== 'AWS::CloudFormation::Stack')
```

Comment:
Why do we filter this?

Reply:
We are only interested in showing the actual errors that we have on resources and not the stacks themselves.